### PR TITLE
fix: Correct ellipsis placement for type_ignore without docstrings

### PR DESCRIPTION
**Pull request by konn**
## Summary
- Fix malformed stub syntax where functions/methods with `type_ignore` attributes but no docstrings generated invalid Python syntax
- Refactor to eliminate code duplication in type_ignore comment generation
- Add comprehensive test cases for edge cases

## Problem
Previously, functions without docstrings but with `#[gen_stub(type_ignore)]` generated malformed stub files:
```python
def func() -> int:  # type: ignore ...
```

## Solution
Now correctly generates valid Python syntax:
```python
def func() -> int: ...  # type: ignore
```

## Changes
- **function.rs**: Refactor Display implementation to place ellipsis before type: ignore comments
- **method.rs**: Apply same fix for methods  
- **examples/pure/src/lib.rs**: Add test functions `test_type_ignore_no_comment_all` and `test_type_ignore_no_comment_specific`
- **Both files**: Eliminate code duplication by calculating type_ignore comment once and reusing

## Test Plan
- [x] Added test cases for functions without docstrings + type_ignore (both catch-all and specific rules)
- [x] Verified existing functionality preserved (functions with docstrings still work)  
- [x] Generated stubs show correct syntax for all type_ignore scenarios
- [x] All existing tests pass (29 total)
- [x] No regressions in stub generation

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: examples/pure/src/lib.rs [hunk 1]
```diff
@@ -684,6 +686,24 @@ fn test_type_ignore_custom() -> i32 {
     42
 }
 
+// NOTE: Doc-comment MUST NOT be added to the next function,
+// as it tests if `type_ignore` without no doccomment is handled correctly;
```
[the remark this answers is not in the document]
Comment (konn) [reply examples/pure/src/lib.rs:690]: That kind of micro-refactoring doesn't pay anything.

File: pyo3-stub-gen/src/generate/method.rs [hunk 2]
```diff
@@ -101,18 +101,29 @@ impl fmt::Display for MethodDef {
...
+            // Add type: ignore comment for methods with docstrings
+            if let Some(comment) = &type_ignore_comment {
+                write!(f, "{}", comment)?;
+            }
             writeln!(f)?;
             let double_indent = format!("{indent}{indent}");
             docstring::write_docstring(f, self.doc, &double_indent)?;
         } else {
-            writeln!(f, " ...")?;
+            write!(f, " ...")?;
```
[the remark this answers is not in the document]
Comment (konn) [reply pyo3-stub-gen/src/generate/method.rs:119]: That kind of micro-refactoring pays nothing.
